Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp

```diff
@@ -1761,6 +1761,34 @@ void ResourceLoadStatisticsStore::requestStorageAccessUnderOpener(DomainInNeedOf
     grantStorageAccessInternal(WTFMove(domainInNeedOfStorageAccess), WTFMove(openerDomain), std::nullopt, openerPageID, StorageAccessPromptWasShown::No, StorageAccessScope::PerPage, canRequestStorageAccessWithoutUserInteraction, [](StorageAccessWasGranted) { });
 }
 
+auto ResourceLoadStatisticsStore::grantStorageAccessPermission(const RegistrableDomain& topFrameDomain, const RegistrableDomain& subFrameDomain) -> std::pair<AddedRecord, std::optional<unsigned>>
+{
+    ASSERT(!RunLoop::isMain());
+
+    auto subFrameStatus = ensureResourceStatisticsForRegistrableDomain(subFrameDomain, "grantStorageAccessPermission"_s);
+    if (!subFrameStatus.second)
+        return subFrameStatus;
+
+    insertDomainRelationshipList(storageAccessUnderTopFrameDomainsQuery, { topFrameDomain }, *subFrameStatus.second);
+    return subFrameStatus;
+}
+
+void ResourceLoadStatisticsStore::revokeStorageAccessPermision(const RegistrableDomain& subFrameDomain)
```

Comment:
`Permision`

Reply:
Also, this doesn't seem right, to revoke storage access for a subframe domain across all top-level frames? This is probably fine right now, specifically for testing, but we need to make sure we don't accidentally use it like this anywhere else.

Oh, and the parameter for this function is not necessarily a subframe domain. The place where you factored this code from was applying it to top-level registrable domains.

Reply:
I'll fix the names. For this function, I was just trying to match how we removed records from this table in clearUserInteraction. But yeah, I think it should be fine for testing.